hphp/runtime/vm/jit/phys-reg-saver.cpp

```diff
@@ -31,6 +31,8 @@ PhysRegSaver::PhysRegSaver(Vout& v, RegSet regs)
   : m_v(v)
   , m_regs(regs)
 {
+  v << vregallow{};
```

Comment:
This should go after the pushes.  You don't know that `push{}` and friends don't lower in a way that creates new Vregs, so it's incorrect to have a `vregallow{}` before everything is saved.